src/pyspiffe/config.py

```diff
@@ -0,0 +1,90 @@
+import os
+import ipaddress
+from urllib.parse import ParseResult, urlparse
+from typing import Dict, List, Tuple
+
+
+SPIFFE_ENDPOINT_SOCKET = 'SPIFFE_ENDPOINT_SOCKET'
+
+
+class ConfigSetter:
+    def __init__(self):
+        self._apply_default_config()
+        self._apply_environment_variables()
+
+    def get_config(self) -> Dict:
+        return self._config
+
+    def _apply_default_config(self):
+        self._config = {
+            SPIFFE_ENDPOINT_SOCKET: None,
+        }
+
+    def _apply_environment_variables(self):
+        endpoint_socket = os.environ.get(SPIFFE_ENDPOINT_SOCKET)
+
+        if endpoint_socket is not None:
+            SpiffeEndpointSocket.validate(endpoint_socket)
+            self._config[SPIFFE_ENDPOINT_SOCKET] = endpoint_socket
+
+
+class SpiffeEndpointSocket:
+    @staticmethod
+    def validate(socket: str):
+        parsed_socket = urlparse(socket)
+
+        if parsed_socket.scheme == '':
+            raise ValueError('SPIFFE endpoint socket: scheme must be set')
+
+        if parsed_socket.scheme == 'unix':
+            SpiffeEndpointSocket.__validate_unix_socket(parsed_socket)
+        elif parsed_socket.scheme == 'tcp':
+            SpiffeEndpointSocket.__validate_tcp_socket(parsed_socket)
+        else:
+            raise ValueError('SPIFFE endpoint socket: unsupported scheme')
```

Comment:
All exception messages on the [spiffe_id module](https://github.com/HewlettPackard/py-spiffe/blob/master/src/pyspiffe/spiffe_id/spiffe_id.py) ends with a final period. I'll think we should be consistent on that.
For now I'll suggest following that format and if required, we can discuss this topic again.